acnportal/acnsim/models/evse.py

```diff
@@ -230,3 +289,35 @@ def _valid_rate(self, pilot, atol=1e-3):
             bool: True if the proposed pilot signal is valid. False otherwise.
         """
         return np.any(np.isclose(pilot, self.allowable_rates, atol=1e-3))
+
+
+    def to_dict(self, context_dict=None):
+        """ Implements BaseSimObj.to_dict. """
+        context_dict, = base.none_to_empty_dict(context_dict)
+        args_dict = super().to_dict(context_dict)
+        args_dict['allowable_rates'] = self.allowable_rates
+        return args_dict
+
+    @classmethod
+    def from_dict(cls, in_dict, context_dict=None, loaded_dict=None, cls_kwargs=None):
+        """ Implements BaseSimObj.from_dict. """
+        context_dict, loaded_dict, cls_kwargs = \
+            base.none_to_empty_dict(context_dict, loaded_dict, cls_kwargs)
+        # TODO: FREVSE constructor doesn't accept args that parent class would
```

Comment:
Add abstract base class for EVSE to avoid different signatures.

Reply:
TODO after PR #53 is accepted.

Reply:
See the updated EVSE class, which uses the new BaseEVSE class to avoid differing signatures in subclasses from their superclass. The serializer and tests have been updated to conform to the new class hierarchy.